crates/mbe/src/syntax_bridge.rs

```diff
@@ -635,9 +635,10 @@ impl<'a> TreeSink for TtTreeSink<'a> {
                     // Mark the range if needed
                     let (text, id) = match leaf {
                         tt::Leaf::Ident(ident) => (ident.text.clone(), ident.id),
-                        tt::Leaf::Punct(punct) => {
-                            (SmolStr::new_inline_from_ascii(1, &[punct.char as u8]), punct.id)
-                        }
+                        tt::Leaf::Punct(punct) => (
+                            std::iter::FromIterator::from_iter(std::iter::once(punct.char)),
```

Comment:
Let's do something more direct here

```
assert(punct.char.is_ascii());
let text = str::from_utf8(slice::from_ref(&(punct.char as u8))).unwrap();
```

Reply:
Thanks, I was indeed not satisfied with instantiating an Iterator just to do that. I hope I understood correctly what you meant because I had to tweak it a little bit.